audiomentations/augmentations/transforms.py

```diff
@@ -374,11 +390,31 @@ def randomize_parameters(self, samples, sample_rate):
     def apply(self, samples, sample_rate):
         num_places_to_shift = self.parameters["num_places_to_shift"]
         shifted_samples = np.roll(samples, num_places_to_shift, axis=-1)
+
+        if type(self.fade_duration) == float:
+            fade_length = int(sample_rate * self.fade_duration)
+        else:
+            fade_length = self.fade_duration
+
         if not self.rollover:
             if num_places_to_shift > 0:
                 shifted_samples[..., :num_places_to_shift] = 0.0
+                if self.fade:
+
+                    # When fading in, the first sample is always 0, that's why
+                    # the fade_length - 1 below.
+
+                    shifted_samples[
+                        ...,
+                        num_places_to_shift - 1 : num_places_to_shift + fade_length - 1,
+                    ] *= np.linspace(0, 1, num=fade_length)
             elif num_places_to_shift < 0:
                 shifted_samples[..., num_places_to_shift:] = 0.0
+                if self.fade:
+                    shifted_samples[
+                        ...,
+                        num_places_to_shift - fade_length + 1 : num_places_to_shift + 1,
```

Comment:
Can you make a unit test that checks if this breaks if fade_length is very large compared to the length of the audio? I'm not sure what happens when a negative index value is less than `-len(the_array)`